core/main/src/broker/rules_engine.rs

```diff
@@ -0,0 +1,205 @@
+        let mut engine = RuleEngine::default();
+        for path in extn_manifest.rules_path.iter() {
+            let path_for_rule = Self::build_path(path, &extn_manifest.default_path);
+            debug!("loading rule {}", path_for_rule);
+            if let Some(p) = Path::new(&path_for_rule).to_str() {
+                if let Ok(contents) = fs::read_to_string(p) {
+                    debug!("Rule content {}", contents);
+                    if let Ok((path, rule_set)) = Self::load_from_content(contents) {
+                        debug!("Rules loaded from path={}", path);
+                        engine.rules.append(rule_set)
+                    } else {
+                        warn!("invalid rule found in path {}", path)
+                    }
+                } else {
+                    warn!("path for the rule is invalid {}", path)
+                }
+            } else {
+                warn!("invalid rule path {}", path)
+            }
+        }
+        engine
+    }
+
+    pub fn load_from_content(contents: String) -> Result<(String, RuleSet), RippleError> {
+        match serde_json::from_str::<RuleSet>(&contents) {
+            Ok(manifest) => Ok((contents, manifest)),
+            Err(err) => {
+                warn!("{:?} could not load rule", err);
+                Err(RippleError::InvalidInput)
+            }
+        }
+    }
+
+    pub fn has_rule(&self, request: &RpcRequest) -> bool {
+        self.rules.rules.contains_key(&request.ctx.method)
+    }
+
+    pub fn get_rule(&self, rpc_request: &RpcRequest) -> Option<Rule> {
+        if let Some(mut rule) = self.rules.rules.get(&rpc_request.method).cloned() {
+            rule.transform.apply_context(rpc_request);
+            return Some(rule);
+        }
+        None
+    }
+}
+
+pub fn jq_compile(input: Value, filter: &str, reference: String) -> Result<Value, RippleError> {
+    debug!("Jq rule {}  input {:?}", filter, input);
+    let start = Utc::now().timestamp_millis();
+    // start out only from core filters,
+    // which do not include filters in the standard library
+    // such as `map`, `select` etc.
+    let mut defs = ParseCtx::new(Vec::new());
+
+    // parse the filter
+    let (f, errs) = jaq_parse::parse(filter, jaq_parse::main());
+    if !errs.is_empty() {
+        error!("Error in rule {:?}", errs);
+        return Err(RippleError::RuleError);
+    }
+
+    // compile the filter in the context of the given definitions
+    let f = defs.compile(f.unwrap());
+    if !defs.errs.is_empty() {
+        error!("Error in rule {}", reference);
+        for (err, _) in defs.errs {
+            error!("reference={} {}", reference, err);
+        }
+        return Err(RippleError::RuleError);
+    }
+
+    let inputs = RcIter::new(core::iter::empty());
+
+    // iterator over the output values
+    let mut out = f.run((Ctx::new([], &inputs), Val::from(input)));
+
+    if let Some(Ok(v)) = out.next() {
+        info!(
```

Comment:
Should this be `debug!`?

Reply:
used as RDK Telemetry marker which is needed for even Prod builds